pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2437,7 +2438,13 @@ private SQLException receiveErrorResponse() throws IOException {
       LOGGER.log(Level.FINEST, " <=BE ErrorMessage({0})", errorMsg.toString());
     }
 
-    PSQLException error = new PSQLException(errorMsg);
+    PSQLException error;
+    if (errorMsg.getMessage() != null && errorMsg.getMessage().endsWith(COPY_ERROR_MESSAGE)) {
```

Comment:
Assuming there is a translated string, will this work?
This will also check for that string for every PSQLException throw, which doesn't look good from a perspective that CopyNotSupportedException might be throw in a 1% of use cases?

Reply:
I'm not sure about the translated string case. There's no translation now but I think you're right about it not working in that situation as the translated one would end up being round tripped. 

As this is only when dealing with errors doubt there's a measurable perf difference for normal usage. Still if there's a way to track the error with a connection level state variable (say a boolean to track the most recent error is COPY related) that'd handle both that and the translation issue.